### DISCUSSION TITLE
Is it a correct use of Nuxt Layers? Or a module would be better?

### DISCUSSION THREAD

**Discussion by educkf** [nuxt/nuxt#27801] [Questions]
Hello!

I have some ecommerces using Nuxt2 that use the same base, but each of those have a lot of duplicated content accross projects, as years ago I have just copy+paste the source code for each new project.

I've been thinking about how to improve that, and I was wondering if using Nuxt Layers would be a good alternative for that.

I would create a Base Nuxt project with base components and composables to handle cart stuff (a beautifull `useCart` composable, for example), payments on server side, rating and these kind of stuff that repeats across projects. Having that built, then I would use it as a Layer on each project, pointing to a release of the layer (to prevent problems of things changing and breaking between projects), and iterate on each project and Base Layer for improvements.

But so far I have absolutelly zero experience with Nuxt Layers... Is this a good/correct use case for the feature? Would be better to have a module or something? I dont know also if in this base layer should I add Pinia, for example... 

I would appreciate some feedback from the community, if anyone have any experience in a situation similar to this one!

Thanks in advance!

**Comment by TheAlexLichter** [comment]
Yes, that totally seems reasonable to use layers for 👍🏻 Also [mentioned a similar case here](https://www.youtube.com/watch?v=fr5yo3aVkfA)

**Reply by educkf** [reply]
Awesome video! Thank you very much. 
I am much more confident now to proceed!
